test/parallel/test-zlib-from-gzip-with-trailing-garbage.js

```diff
@@ -42,8 +49,11 @@ data = Buffer.concat([
   Buffer([0x1f, 0x8b, 0xff, 0xff])
 ]);
 
-assert.throws(() => zlib.gunzipSync(data));
+assert.throws(
+  () => zlib.gunzipSync(data),
+  /^Error: unknown compression method$/
+);
 
-zlib.gunzip(data, common.mustCall((err, result) => {
+zlib.gunzip(data, common.mustCall((err) => {
```

Comment:
Same suggestions here as above. Restore `result`, add an assertion to confirm it is `undefined`, and do some more rigorous checking on `err`.